### PR TITLE
[29] Export tasks in pdf

### PR DESCRIPTION
**Pull request by Amir-Gaifullin**
### Summary

Add pdf format for `task`

### How it works

Снимок экрана 2022-10-14 в 18 07 50

### Test plan
N/A
### Review notes

While reviewing pull-request (especially when it's your pull-request),
please make sure that:

- you understand what problem is solved by PR and how is it solved
- new tests are in place, no redundant tests
- DB schema changes reflect new migrations
- newly introduced DB fields have indexes and constraints
- there are no missed files (migrations, view templates)
- required ENV variables added and described in `.env.example` and added to Heroku
- associated Heroku review app works correctly with introduced changes

### Deploy notes

Notes regarding deployment the contained body of work.
These should note any db migrations, ENV variables, services, scripts, etc.

### CODE REVIEW

File: app/controllers/tasks_controller.rb [hunk 1]
```diff
@@ -11,6 +11,16 @@ def index
   def show
     @comments = Comment.where(task_id: @task.id)
     @comment = Comment.new
+    respond_to do |format|
+      format.html
+      format.pdf do
+        render pdf: [@task.id, @task.title].join('-'),
+               template: 'tasks/task',
+               formats: [:html],
+               disposition: :inline,
```
Comment (RasimKhusaenov) [inline app/controllers/tasks_controller.rb:20]: It is `inline` by default

File: app/views/layouts/pdf.html.erb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    <meta content="text/html; charset=UTF-8" http-equiv="Content-Type"/>
+    <meta name="viewport" content="width=device-width,initial-scale=1">
+    <%= csrf_meta_tags %>
+    <%= csp_meta_tag %>
+    <%= wicked_pdf_stylesheet_link_tag "pdf" %>
+  </head>
+  <body>
+    <div id="header">
+      <!--= wicked_pdf_image_tag 'thumbnail.png', height: "30", width: "auto"-->
+    </div>
```
Comment (RasimKhusaenov) [inline app/views/layouts/pdf.html.erb:14]: Why?)

File: app/views/tasks/task.html.slim [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  strong Description:
+  = @task.description
+
+p
+  strong Status:
+  = @task.status
+
+p
+  strong Project:
+  = link_to @task.project.name, @task.project
```
Comment (RasimKhusaenov) [inline app/views/tasks/task.html.slim:23]: Maybe it's better to define @project in the controller?

File: config/initializers/wicked_pdf.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+WickedPdf.config ||= {}
+WickedPdf.config.merge!({
+  layout: "pdf.html.erb",
+  orientation: "Landscape", # Portrait
+  page_size: "A4",
```
Comment (RasimKhusaenov) [inline config/initializers/wicked_pdf.rb:5]: It is default value

File: config/initializers/wicked_pdf.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+WickedPdf.config ||= {}
+WickedPdf.config.merge!({
+  layout: "pdf.html.erb",
+  orientation: "Landscape", # Portrait
```
Comment (RasimKhusaenov) [inline config/initializers/wicked_pdf.rb:4]: Landscape != Portrait

File: spec/examples.txt [hunk 6]
```diff
@@ -1,21 +0,0 @@
-example_id                                      | status  | run_time        |
```
Comment (RasimKhusaenov) [inline spec/examples.txt:1]: I think, you should add this file to `.gitignore`

Comment (Amir-Gaifullin) [reply spec/examples.txt:1]: image

File: app/controllers/tasks_controller.rb (hunk 1 above)
Comment (Amir-Gaifullin) [reply app/controllers/tasks_controller.rb:20]: copy-paste code

File: config/initializers/wicked_pdf.rb (hunk 4 above)
Comment (Amir-Gaifullin) [reply config/initializers/wicked_pdf.rb:5]: copy-paste code

File: config/initializers/wicked_pdf.rb [hunk 7]
```diff
@@ -0,0 +1,9 @@
...
+  page_size: "A4",
+  lowquality: true,
+  zoom: 1,
+  dpi: 75
```
[the remark this answers is not in the document]
Comment (Amir-Gaifullin) [reply config/initializers/wicked_pdf.rb:8]: copy-paste

File: spec/examples.txt (hunk 6 above)
Comment (Amir-Gaifullin) [reply spec/examples.txt:1]: i have it

**Pull request merged**